server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -412,14 +425,24 @@ public RemoteClusterStateManifestInfo writeIncrementalMetadata(
         uploadedMetadataResults.uploadedIndicesRoutingMetadata = remoteRoutingTableService.getAllUploadedIndicesRouting(
             previousManifest,
             uploadedMetadataResults.uploadedIndicesRoutingMetadata,
-            routingTableDiff.getDeletes()
+            routingTableIncrementalDiff.getDeletes()
         );
 
+        ClusterStateDiffManifest clusterStateDiffManifest = new ClusterStateDiffManifest(
+            clusterState,
+            previousClusterState,
+            routingTableIncrementalDiff
+        );
+        if (uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata != null) {
+            clusterStateDiffManifest.setIndicesRoutingDiffPath(
+                uploadedMetadataResults.uploadedIndicesRoutingDiffMetadata.getUploadedFilename()
+            );
+        }
```

Comment:
Should we have this as a constructor param only?

Reply:
Added it as part of constructor.